### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by mcelicalderon**

### CODE REVIEW

File: .environment [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export DB_USERNAME=postgres
 export DB_PASSWORD=postgres
-export JDBC_URL=jdbc:postgresql://localhost:5432/airtng-servlets
\ No newline at end of file
+export JDBC_URL=jdbc:postgresql://localhost:5432/ivr-recording
```
Comment (acamino) [inline .environment]: Nice catch!

File: README.md [hunk 2]
```diff
@@ -18,18 +22,18 @@ Twilio and Java Servlets.
...
-    $ createdb airtng-servlets
+    $ createdb ivr-recording
 
     ```
   _The application uses PostgreSQL as the persistence layer. If you
   don't have it already, you should install it. The easiest way is by
   using [Postgres.app](http://postgresapp.com/)._
 
-1. Edit the sample configuration file `.environment` to match your database       configuration:
+1. Edit the sample configuration file `.environment` to match your database configuration:
```
Comment (acamino) [inline README.md]: **Not for this one**, but I'm thinking about using always `.env.example` as the canonical way to group the environment variables. Of course this approach will includes adding the `.env` to the _.gitignore_.

What do you think?

File: src/main/java/com/twilio/ivrrecording/servlet/WebAppServlet.java [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
-    protected void respondTwiML(HttpServletResponse response, TwiMLResponse twiMLResponse) throws IOException {
-        response.setContentType("text/xml");
-        response.getWriter().write(twiMLResponse.toXML());
+  protected void respondTwiML(HttpServletResponse response, VoiceResponse twiMLResponse)
+      throws IOException {
+    response.setContentType("text/xml");
+    try {
+      response.getWriter().write(twiMLResponse.toXml());
+    } catch (TwiMLException e) {
+      System.out.println("Unable to create TwiML");
```
Comment (acamino) [inline src/main/java/com/twilio/ivrrecording/servlet/WebAppServlet.java]: I would prefer to re-throw the exception. ;)

File: src/main/java/com/twilio/ivrrecording/servlet/agent/CallServlet.java [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
+        .build();
+
+    Say say2 = new Say.Builder("No record received. Goodbye")
+        .language(Say.Language.EN_GB)
+        .voice(Say.Voice.ALICE)
+        .build();
+
+    Hangup hangup = new Hangup();
+
+    VoiceResponse voiceResponse = new VoiceResponse.Builder()
```
Comment (acamino) [inline src/main/java/com/twilio/ivrrecording/servlet/agent/CallServlet.java]: Although this is a perfectly valid approach, I think we can increase the readability of this piece of code by removing variables like `say1`, `say2` and so on, something like this:

```java
VoiceResponse voiceResponse = new VoiceResponse.Builder()
        .say(new Say.Builder(
                "It appears that no agent is available. " + "Please leave a message after the beep")
                .language(Say.Language.EN_GB)
                .voice(Say.Voice.ALICE)
                .build())
        .record(new Record.Builder()
                .maxLength(20)
                .action("/agents/hangup")
                .transcribeCallback(String.format("/records/create?agentId=%s", agentId))
                .build())
        .say(new Say.Builder("No record received. Goodbye")
                .language(Say.Language.EN_GB)
                .voice(Say.Voice.ALICE)
                .build())
        .hangup(new Hangup())
        .build();
```

What do you think?

Comment (mcelicalderon) [inline src/main/java/com/twilio/ivrrecording/servlet/agent/CallServlet.java]: Hey @mosampaio , already discussed with @acamino . What do you think?

Comment (mosampaio) [inline src/main/java/com/twilio/ivrrecording/servlet/agent/CallServlet.java]: I prefer without the variables as well.

Comment (acamino) [inline src/main/java/com/twilio/ivrrecording/servlet/agent/CallServlet.java]: Just to clarify, I prefer my proposed version. 😉  Mario likes the one which extract variables like `say1`.

**Pull request merged**